src/common/SmartUrlInput.js

```diff
@@ -45,7 +49,7 @@ type Props = $ReadOnly<{|
    * it appears not to contain an explicit domain.
    */
   defaultDomain: string,
-  navigation: NavigationScreenProp<mixed>,
+  navigation: NavigationScreenProp<NavigationState>,
```

Comment:
Not a big deal, but I suspect you can make this change *after* the "Tell all screens on AppNavigator" commit (and include the `RealmScreen` change in that one.) If you do, then I think the story to tell in this one's commit message becomes simpler, because no foreshadowing is necessary.

Reply:
I get an error if I leave this as `mixed` with the `RealmScreen` change applied:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/start/RealmScreen.js:121:10

Cannot create SmartUrlInput element because in property navigation:
 • Either cannot create exact type from mixed [1].
 • Or property getParam is missing in object type [2] but exists in NavigationScreenProp [3].

        src/start/RealmScreen.js
        118│         shouldShowLoadingBanner={false}
        119│       >
        120│         <Label text="Enter your Zulip server URL:" />
        121│         <SmartUrlInput
        122│           style={styles.input}
        123│           navigation={navigation}
        124│           defaultProtocol="https://"
        125│           defaultOrganization="your-org"
        126│           defaultDomain="zulipchat.com"
        127│           defaultValue={initialRealm}
        128│           onChangeText={this.handleRealmChange}
        129│           onSubmitEditing={this.tryRealm}
        130│           enablesReturnKeyAutomatically
        131│         />
        132│         {error !== null ? (
        133│           <ErrorMsg error={error} />
        134│         ) : (

        flow-typed/npm/react-navigation-stack_v1.x.x.js
    [2] 595│     & {
        596│         pop: (n?: number, params?: {| immediate?: boolean |}) => boolean,
        597│         popToTop: (params?: {| immediate?: boolean |}) => boolean,
        598│         push: (
        599│           routeName: string,
        600│           params?: NavigationParams,
        601│           action?: NavigationNavigateAction
        602│         ) => boolean,
        603│         replace: (
        604│           routeName: string,
        605│           params?: NavigationParams,
        606│           action?: NavigationNavigateAction
        607│         ) => boolean,
        608│         reset: (actions: NavigationAction[], index: number) => boolean,
        609│         ...
        610│       };

        src/common/SmartUrlInput.js
 [3][1]  52│   navigation: NavigationScreenProp<mixed>,
```

Reply:
I could do a temporary `$FlowFixMe`, but I wonder if that would mean roughly the same amount of explanation that's already in this commit?

Reply:
Huh. That's a bit puzzling to me: it feels like it should be perfectly fine, and also when I look at the actual type definitions it seems like it should be accepted. In particular, the new type `NavigationScreenProp<NavigationState>` should be a subtype of the old type `NavigationScreenProp<mixed>` -- `NavigationState` is a subtype of `mixed` because everything is, and `NavigationScreenProp` is declared in the react-navigation libdef as covariant (with a `+`) in its type parameter:
```js
  declare export type NavigationScreenProp<+S> = {
```
so S a subtype of T should make `NavigationScreenProp<S>` a subtype of `NavigationScreenProp<T>`.

But not worth trying to debug further, since you've already made a fine workaround.